### PR TITLE
Rose user guide site and user configuration

### PR DESCRIPTION
**Pull request by oliver-sanders**
~**Built on #2161, hold review until #2161 has been merged.**~
**Ready for review.**

* Add a system to auto document rose configuration files using RST formatted comments and the rose domain.
* Use the auto-documenter on the ``etc/rose.conf.example`` file to generate the site/user api reference material.

**TODO**

* Check default settings.
* Ensure example configuration file is complete
* Improve comments.

**Scope For Future Work**

* Specify options in ``:opt:`` sections.
* Include certain metadata settings if defined e.g. ``type`` and ``length``.

@matthewrmshin This may need a little work, feel free to raise a PR against this branch for larger changes. ~Assign a second reviewer once you are happy.~

### CODE REVIEW

File: etc/rose.conf.example [hunk 1]
```diff
@@ -1,444 +1,445 @@
...
-## Default access-list setting on suite creation
-#  access-list-default=USER-ID ...
-## E.g. to grant access to all users by default:
-#  access-list-default=*
-[rosie-vc]
+#    access-list-default=*
+access-list-default=USER-ID ...
+
+
+# TODO - write this section
```
Comment (oliver-sanders) [inline etc/rose.conf.example]: Missing ``[rosie-ws]`` section or obsolete installation instructions?

Comment (matthewrmshin) [reply etc/rose.conf.example]: Section became `[rosie-disco]`.

**Comment by matthewrmshin** [comment]
Feedback @ oliver-sanders/rose#17

**Comment by sadielbartholomew** [comment]
I've opened a PR for small amendments to 'polish' the Site and User Configuration page, but otherwise it looks really good. The auto-documentation feature seems like it could be really useful. Docs build cleanly. Noting conversation in person RE default syntax highlighting as 'rose' appears to usually be more applicable and certainly applies better for code in the Site and User Configuration, if not in all cases.

**Comment by oliver-sanders** [comment]
>  default syntax highlighting style as 'rose'

Done, though I stuck this commit in the other PR 80e1683

**Pull request merged**